hoomd/pytest/test_simulation.py

```diff
@@ -197,6 +197,31 @@ def test_state_from_gsd(simulation_factory, get_snapshot,
         assert box == sim.state.box
         assert_equivalent_snapshots(snap, sim.state.snapshot)
 
+@skip_gsd
+def test_state_from_gsd_snapshot(simulation_factory, get_snapshot,
+                                 device, state_args, tmp_path):
+    snap_params, nsteps = state_args
+
+    sim = simulation_factory(get_snapshot(n=snap_params[0],
+                                          particle_types=snap_params[1]))
+    snap = sim.state.snapshot
+    snap = make_gsd_snapshot(snap)
+    gsd_snapshot_list = [snap]
+    box = sim.state.box
+    for _ in range(1, nsteps):
+        particle_type = np.random.choice(snap_params[1])
+        snap = update_positions(sim.state.snapshot)
+        set_types(snap, random_inds(snap_params[0]),
+                  snap_params[1], particle_type)
+        snap = make_gsd_snapshot(snap)
+        gsd_snapshot_list.append(snap)
```

Comment:
I agree that this test could be shorter. However, it is written in the same style as the other tests in this file so I think it is OK for now. Any future clean up in this file can clean up all the tests in a similar way.